packages/wrangler/src/deployment-bundle/worker.ts

```diff
@@ -336,6 +336,10 @@ export interface CfWorkerInit {
 	tail_consumers: CfTailConsumer[] | undefined;
 	limits: CfUserLimits | undefined;
 	annotations?: Record<string, string | undefined>;
+	experimental_assets?: {
+		jwt: string | undefined;
+		staticAssetsOnly: boolean | undefined;
```

Comment:
I probably missed the discussion but could we not have `main` be optionally undefined. So if there is no `main` but there is `experimental_assets` then one can infer that this is static assets only?

There would be an assertion that at least on of `main` and `experimental_assets` is defined.

Reply:
This is the end goal but waiting for the no-op-asset-worker to be removed first :)